assembl/static2/js/app/components/debate/common/announcement.spec.jsx

```diff
@@ -0,0 +1,240 @@
+import React from 'react';
+import renderer from 'react-test-renderer'; // eslint-disable-line
+
+import Announcement, {
+  createDoughnutElements,
+  createTooltip,
+  getColumnInfos,
+  getSentimentsCount,
+  dirtySplitHack
+} from './announcement';
+
+describe('Announcement component', () => {
+  const announcementContent = {
+    body: '<p>Bonjour à tous</p><p>Nous allons mettre un deuxieme paragraphe</p><p>Et un peu de <strong>texte en gras</strong>'
+  };
+  const fakePost1 = {
+    __typename: 'PostEdge',
+    node: {
+      __typename: 'Post',
+      id: '9706',
+      parentId: 'null',
+      creationDate: '2018-02-21T18:07:49.630164+00:00',
+      publicationState: 'PUBLISHED',
+      originalLocale: 'fr',
+      sentimentCounts: { like: 2, disagree: 1, dontUnderstand: 0, moreInfo: 1, __typename: 'SentimentCounts' }
+    }
+  };
+  const fakePost2 = {
+    __typename: 'PostEdge',
+    node: {
+      __typename: 'Post',
+      id: '9706',
+      parentId: '9706',
+      creationDate: '2018-02-21T18:08:14.590132+00:00',
+      publicationState: 'PUBLISHED',
+      originalLocale: 'fr',
+      sentimentCounts: { like: 1, disagree: 1, dontUnderstand: 9, moreInfo: 0, __typename: 'SentimentCounts' }
+    }
+  };
+  const fakeThreadPosts = { edges: [fakePost1, fakePost2] };
+  describe('<Announcement/> on a thread', () => {
+    it('should render an announcement with the content associated to a thread idea', () => {
+      const threadIdea = {
+        id: '1234',
+        messageColumns: [],
+        messageViewOverride: null,
+        numContributors: 2,
+        numPosts: 2,
+        posts: fakeThreadPosts,
+        __typename: 'Idea'
+      };
+      const announcementProps = {
+        announcementContent: announcementContent,
+        idea: threadIdea,
+        isMultiColumns: false
+      };
+      const component = renderer.create(<Announcement {...announcementProps} />);
+      const tree = component.toJSON();
+      expect(tree).toMatchSnapshot();
```

Comment:
From my point of view, having something matching a snapshot don't really test an expected result.